enumerables.rb

```diff
@@ -0,0 +1,137 @@
+module Enumerable
+  def my_each
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i])
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    arr = self
+    if block_given?
+      while i < arr.count
+        yield(arr[i], i)
+        i += 1
+      end
+    else
+      arr
+    end
+  end
+
+  def my_select
+    i = 0
+    to_change = []
+    arr = self
+    if block_given?
+      while i < arr.count
+        to_change << arr[i] if yield(arr[i])
+        i += 1
+      end
+      to_change
+    else
+      arr
+    end
+  end
+
+  def my_any?(arg = nil)
+    arr = self
+    i = 0
+    to_change = false
+    while i < arr.count
+      if !block_given?
+        to_change = true if include?(true) && arg.nil?
+        to_change = true if arr[i] == arg && arg.respond_to?(:to_i)
+        to_change = true if arr[i].match(arg) && arg.is_a?(Regexp)
+        to_change = true if arr[i].instance_of?(arg) && arg.respond_to?(:class)
+      else
+        to_change = false unless yield(arr[i])
+      end
+      i += 1
+    end
+    to_change
+  end
+
+  def my_all?
+    i = 0
+    to_change = true
+    arr = self
+    while i < arr.count
+      to_change = false unless yield(arr[i])
+      i += 1
+    end
+    to_change
+  end
+
+  def my_none?
```

Comment:
- [ ] my_none should still work without a block or an argument. It returns true if all items are falsy in this case.
- [ ] Make sure to add conditions that take into account getting a Regex, Class, or another pattern as an argument.